phpBB/phpbb/profilefields/type/type_interface.php

```diff
@@ -200,11 +208,11 @@ public function prepare_hidden_fields($step, $key, $action, &$field_data);
 	/**
 	* Allows assigning of additional template variables
 	*
-	* @param array	$template_vars	Template variables we are going to assign
+	* @param string	$action			Currently performed action (create|edit)
 	* @param array	$field_data		Array with data for this field
-	* @return null
+	* @return array	with the acp options for step 1
 	*/
-	public function display_options(&$template_vars, &$field_data);
+	public function display_options($action, &$field_data);
```

Comment:
Ok, I can make a proposal that is 100% BC, but is a kind of spaguetti code. Stay tuned.